frontend/src/views/WheresMyGene/components/HeatMap/components/XAxisChart/index.tsx

```diff
@@ -18,19 +21,25 @@ import {
   XAxisLabel,
   GeneButtonStyle,
   XAxisGeneName,
+  InfoButtonWrapper,
 } from "./style";
-
+import { EXCLUDE_IN_SCREENSHOT_CLASS_NAME } from "../../../GeneSearchBar/components/SaveExport";
+import { StyledImage } from "../YAxisChart/style";
+import InfoSVG from "../YAxisChart/icons/info-sign-icon.svg";
 interface Props {
   geneNames: string[];
+  generateGeneInfo: (gene: string) => void;
 }
 
 function GeneButton({
   geneName,
   genesToDelete,
+  generateGeneInfo,
 }: {
   geneName: string;
   genesToDelete: string[];
   handleGeneClick: (gene: string) => void;
+  generateGeneInfo: (gene: string) => void;
```

Comment:
passing in generateGeneInfo function to generate gene info when icon button is clicked on the x axis